app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -128,25 +128,47 @@ export default class RnsResolver extends RnsJsDelegate {
     });
   }
 
-  // this.rifConfig.rns.contracts.multiChainResolver This is the path to a multicahin address in the config
-  // TODO: Rodrigo
-  getNetoworksForResolvers (address) {
+  /**
+   * Returns an array of network addresses for a given domain
+   * @param domainNameResolver DomainName with the .rsk extension
+   * @returns {Promise<unknown>}
+   */
+  getNetworksForResolvers (domainNameResolver) {
     return new Promise((resolve, reject) => {
+      // TODO: Rodrigo
       // This event is different, it deppends if we're on the chain of RSK, or not, to develop purposes i'll use only ChainAddrChanged, it need to differentiate it from one to another
-      console.debug('INICIO');
-
-      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash('testing.rsk')}, {fromBlock: 0, toBlock: 'latest'});
+      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainNameResolver)}, {fromBlock: 0, toBlock: 'latest'});
       myEvent.get(function(error, result){
-        console.debug('RESULT ===================', result);
-        console.debug('ERROR ===================', error);
+        if (error) {
+          console.debug('Error when trying to get netoworks for resolver', error);
+          reject(error);
+        }
+        const arrChains = [];
+        result.forEach(network => {
+          const networkToPush = new ResolverNetwork(network.args.chain, network.args.addr)
+          const index = arrChains.findIndex((e) => e.chain === networkToPush.chain);
+          if (index === -1) {
+            arrChains.push(networkToPush);
+          } else {
+            arrChains[index] = networkToPush;
+          }
+        });
+        console.debug('getNetworksForResolvers success', arrChains);
+        resolve(arrChains);
       });
-      console.debug('FIN');
     });
   }
 
-  setNetoworksForResolver () {
+  /**
+   * Calls the contract and sets a new network calling the setChainAddr function in the contract of multichainresolver
+   * @param domainNameResolver domainNameResolver DomainName with the .rsk extension
+   * @param chain
+   * @param networkAddress
+   * @returns {Promise<unknown>}
+   */
+  setNetoworksForResolver (domainNameResolver, chain, networkAddress) {
     return new Promise((resolve, reject) => {
-      this.send(this.multiChainresolverContractInstance, 'setChainAddr', [namehash.hash('lebron.rsk'), '0x80000069', '0x123456789'])
+      this.send(this.multiChainresolverContractInstance, 'setChainAddr', [namehash.hash(domainNameResolver), chain, networkAddress])
```

Comment:
Why `domainNameResolver`? isnt that parameter the bytes32 node of the domain? Its just the `domainName`, is not related with the resolver at all

Reply:
Changed in previous commit!